Plugin infrastructure

todvora opened the pull request:
onConfig and outputFormat hooks for plugins, config read unified. See also #92.

JamesCropcho commented:
This is all very exciting, man! Don't you think it's long, long overdue for a pretty big version bump ;) @todvora?

todvora commented:
Thanks! You are right about the version number. It is now changed from 1.4.1 to 1.5.0. I added functionality in a backwards-compatible manner and thus the minor version should be incremented (due to [semver](http://semver.org/)).

JamesCropcho commented:
Happily merged, man! The start of a new epoch.
